devices/Adxl343/ADXL343.cs

```diff
@@ -0,0 +1,1821 @@
+        /// Register 0x33 hold the output data for the x-axis, Register 0x34 and
+        /// Register 0x35 hold the output data for the y-axis, and Register 0x36
+        /// and Register 0x37 hold the output data for the z-axis.The output
+        /// data is twos complement, with DATAx0 as the least significant byte
+        /// and DATAx1 as the most significant byte, where x represent X,
+        /// Y, or Z. The DATA_FORMAT register (Address 0x31) controls the
+        /// format of the data.It is recommended that a multiple-byte read of all
+        /// registers be performed to prevent a change in data between reads
+        /// of sequential registers.
+        /// </summary>
+        /// <param name="accel">Acceleration.</param>
+        /// <returns>True if Full Transfer result, false if any other result.</returns>
+        public bool GetAcceleration(ref Vector3 accel)
+        {
+            SpanByte readBuf = new byte[7];
+            var res = i2c.WriteByte((byte)Register.X0);
+            res = i2c.Read(readBuf);
+
+            if (res.Status == I2cTransferStatus.FullTransfer)
+            {
+                short accelerationX = BinaryPrimitives.ReadInt16LittleEndian(readBuf.Slice(0, 2));
+                short accelerationY = BinaryPrimitives.ReadInt16LittleEndian(readBuf.Slice(2, 2));
+                short accelerationZ = BinaryPrimitives.ReadInt16LittleEndian(readBuf.Slice(4, 2));
+
+                accel.X = (float)accelerationX;
+                accel.Y = (float)accelerationY;
+                accel.Z = (float)accelerationZ;
+               
+                return true;
+            }
+
+            return false;
+        }
+
+        #endregion
+
+        #region FIFO Control
+
+        /// <summary>
+        /// These bits set the FIFO mode, as described in the Table
+        /// FIFO Modes
+        ///     D7      D6      Mode        Function
+        ///     0       0       Bypass      FIFO is bypassed.
+        ///     0       1       FIFO        FIFO collects up to 32 values and then stops collecting data, collecting new data only when FIFO is not
+        ///                                 full.
+        ///     1       0       Stream      FIFO holds the last 32 data values. When FIFO is full,
+        ///                                 the oldest data is overwritten with newer data.
+        ///     1       1       Trigger     When triggered by the trigger bit, FIFO holds the last
+        ///                                 data samples before the trigger event and then contin-
+        ///                                 ues to collect data until full. New data is collected only
+        ///                                 when FIFO is not full.
+        /// Trigger Bit
+        /// A value of 0 in the trigger bit links the trigger event of trigger mode
+        /// to INT1, and a value of 1 links the trigger event to INT2.
+        /// Samples Bits
```

Comment:
what are the valid values for samples? You should check that in the set at least and if few, then an enum will be the good way to go